tests/Game.class.test.js

```diff
@@ -0,0 +1,916 @@
+
+        game2048.start();
+        game2048.moveDown();
+
+        const state = game2048.getState();
+        const stateTransposed = transposeState(state);
+        const initialStateTransposed = transposeState(initialState);
+
+        stateTransposed.forEach((column, colIndex) => {
+          const initialColumn = initialStateTransposed[colIndex];
+          const nonEmptyCells = initialColumn.filter((cell) => cell !== 0);
+          const cursor = nonEmptyCells[0];
+
+          expect(checkIsCorrectlyAligned(column, 'end', cursor)).toBe(true);
+
+          nonEmptyCells.forEach((cell) => {
+            expect(column.includes(cell)).toBe(true);
+          });
+        });
+      });
+
+      it('should merge cells with the same value down', () => {
+        const initialState = [
+          [16, 0, 0, 0],
+          [0, 0, 8, 8],
+          [0, 8, 8, 0],
+          [8, 8, 0, 16],
+        ];
+
+        const game2048 = new Game(cloneState(initialState));
+
+        game2048.start();
+        game2048.moveDown();
+
+        const state = game2048.getState();
+        const stateTransposed = transposeState(state);
+
+        stateTransposed.forEach((column) => {
+          expect(column.includes(16)).toBe(true);
+        });
+      });
+
+      it('should merge multiple cells with the same value down', () => {
+        const initialState = [
+          [8, 16, 32, 16],
+          [8, 16, 32, 16],
+          [16, 8, 8, 16],
+          [16, 8, 8, 16],
+        ];
+
+        const game2048 = new Game(cloneState(initialState));
+
+        game2048.start();
+        game2048.moveDown();
+
+        const getCell = getCellFromState(game2048);
+
+        expect(getCell(2, 0)).toBe(16);
+        expect(getCell(2, 1)).toBe(32);
+        expect(getCell(2, 2)).toBe(64);
+        expect(getCell(2, 3)).toBe(32);
+        expect(getCell(3, 0)).toBe(32);
+        expect(getCell(3, 1)).toBe(16);
+        expect(getCell(3, 2)).toBe(16);
+        expect(getCell(3, 3)).toBe(32);
+      });
+    });
+  });
+
+  describe('game status', () => {
+    it('should be a "playing" when board is full but moves are still possible', () => {
+      const game2048 = new Game([
+        [128, 128, 0, 8],
+        [16, 8, 16, 32],
+        [8, 16, 32, 64],
+        [16, 32, 0, 128],
+      ]);
+
+      game2048.start();
+
+      expect(game2048.getStatus()).toBe('playing');
+    });
+
+    it('should be a "win" when a 2048 tile is created', () => {
+      const game2048 = new Game([
+        [0, 0, 0, 0],
+        [0, 0, 1024, 1024],
+        [0, 0, 0, 0],
+        [0, 0, 0, 0],
+      ]);
+
+      game2048.start();
+      game2048.moveRight();
+
+      expect(game2048.getStatus()).toBe('win');
+    });
+
+    it('should be a "lose" when no more moves are possible', () => {
+      const game2048 = new Game([
+        [128, 128, 0, 8],
+        [16, 8, 16, 32],
+        [8, 16, 32, 64],
+        [16, 32, 0, 128],
+      ]);
+
+      game2048.start();
+      game2048.moveLeft();
```

Comment:
I think that 2 new items will appear after the start 
So the game should be in a `lose` status before the `moveLeft`

Reply:
There are two empty cells, so after `start()` they will be filled with 2 cells (value 2 or 4). But one move will be possible - to the left. It merges 128 + 128. After this only one cell will be available - 1st row, 4th cell. It will be filled again with 2 or 4. But the board will be blocked in that state, so game is lost.

Reply:
ok
missed that move